### PR TITLE
[CLB-21]Test PR

### PR DESCRIPTION
**Pull request by Zola987**

### CODE REVIEW

**Review by Mbotunjac — APPROVED** [review]

**Comment by Mbotunjac** [comment]
looks good

File: src/App.tsx [hunk 1]
```diff
@@ -9,21 +9,78 @@ import TestVS from '@pages/TestVS';
...
+                    }
+                />
+                <Route
+                    path="/header"
+                    element={
+                        <PrivateRoute>
+                            <Header />
+                        </PrivateRoute>
+                    }
+                />
```
Comment (kimnovak) [inline src/App.tsx:40]:
```suggestion
                <Route
                    path="/header"
                    element={
                        <ProtectedRoute>
                             <PrivateRoute>
                                 <Header />
                            </PrivateRoute>
                        </ProtectedRoute>
                    }
                />
```

File: src/App.tsx [hunk 2]
```diff
@@ -9,21 +9,78 @@ import TestVS from '@pages/TestVS';
...
+                <Route
+                    path="/"
+                    element={
+                        <PrivateRoute>
+                            <Home />
+                        </PrivateRoute>
+                    }
+                />
```
Comment (kimnovak) [inline src/App.tsx:56]:
```suggestion
                <Route
                    path="/"
                    element={
                        <ProtectedRoute>
                            <PrivateRoute>
                                <Home />
                            </PrivateRoute>
                        </ProtectedRoute>
                    }
                />
```

File: src/App.tsx [hunk 3]
```diff
@@ -9,21 +9,78 @@ import TestVS from '@pages/TestVS';
...
+                    element={
+                        <PrivateRoute>
+                            <Test />
+                        </PrivateRoute>
+                    }
+                />
                 <Route path="/testsm" element={<TestSM />} />
                 <Route path="/testvs" element={<TestVS />} />
                 <Route path="/testbb" element={<TestBB />} />
                 <Route path="/testss" element={<TestSS />} />
```
Comment (kimnovak) [inline src/App.tsx]:
```suggestion
                <Route path="/testss" element={<TestSS />} />
                <Route
                    path="/timesheet"
                    element={
                        <ProtectedRoute>
                            <Timesheet />
                        </ProtectedRoute>
                     }
                 />
```

File: src/components/Login/index.tsx [hunk 4]
```diff
@@ -1,16 +1,42 @@
...
+        email: '',
+        password: '',
+    });
+    type userDetails = {
+        email: string;
+        password: string;
+    };
+
+    function login(details: userDetails): void {
+        console.log(details);
```
Comment (kimnovak) [inline src/components/Login/index.tsx]:
```suggestion
```

File: src/components/Login/index.tsx [hunk 5]
```diff
@@ -1,16 +1,42 @@
...
 
-    function validateForm() {
-        return email && password;
+        if (details.email === 'user@user.com' && details.password === 'user') {
+            console.log('logged in');
```
Comment (kimnovak) [inline src/components/Login/index.tsx:20]:
```suggestion
```

File: src/components/Login/index.tsx [hunk 6]
```diff
@@ -1,16 +1,42 @@
...
+            localStorage.setItem('isLogged', 'true');
+        } else if (
+            details.email === 'admin@admin.com' &&
+            details.password === 'admin'
+        ) {
+            // localStorage.setItem('isLogged', 'false')
+
+            console.log('logged in');
```
Comment (kimnovak) [inline src/components/Login/index.tsx]:
```suggestion
```

File: src/App.tsx [hunk 7]
```diff
@@ -9,21 +9,78 @@ import TestVS from '@pages/TestVS';
 import TestSM from '@pages/TestSM';
 import Header from '@components/Header/Header';
 import LoginPage from '@pages/LoginPage';
-import Timesheet from '@pages/Timesheet';
+import Test from '@components/PrivateRoute/Test';
+
```
Comment (kimnovak) [inline src/App.tsx:14]:
```suggestion
```

File: src/components/Login/index.tsx [hunk 8]
```diff
@@ -1,16 +1,42 @@
...
+        } else {
+            localStorage.setItem('isLogged', 'false');
+            localStorage.setItem('admin', 'false');
+        }
     }
 
     function handleSubmit(event: FormEvent<HTMLFormElement>) {
         event.preventDefault();
+
+        login(details);
```
Comment (kimnovak) [inline src/components/Login/index.tsx:40]: after logging in we should redirect the user to the homepage
You can achieve it by adding this at the top of this component
```
const navigate = useNavigate();
```
and then calling `navigate('/')` after login

File: src/pages/TestBB.tsx [hunk 9]
```diff
@@ -15,6 +15,7 @@ function TestBB() {
                 startDate={'01-01-2021'}
                 endDate={'N/A'}
             />
+            {/* <Timesheet /> */}
```
Comment (kimnovak) [inline src/pages/TestBB.tsx]:
```suggestion
```

File: src/redux/actions/actionTypes.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+const actionTypes = {
+    ADD: 'ADD',
+    SUB: 'SUB',
+};
+
+export default actionTypes;
```
Comment (kimnovak) [inline src/redux/actions/actionTypes.ts]: src/redux got renamed to src/reduxStore, you should delete the old directory

File: src/redux/reducers/index.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import counterReducer from '@reduxStore/reducers/counterReducer';
+import { combineReducers } from 'redux';
+
+const allReducers = combineReducers({
+    count: counterReducer,
+});
+
+export default allReducers;
+export type RootState = ReturnType<typeof allReducers>;
```
Comment (kimnovak) [inline src/redux/reducers/index.ts]: src/redux got renamed to src/reduxStore, you should delete the old directory

File: src/redux/store/store.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import allReducers from '@reduxStore/reducers/index';
+import { createStore, applyMiddleware } from 'redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
+
+const composedEnhancer = composeWithDevTools(applyMiddleware());
+
+const store = createStore(allReducers, composedEnhancer);
+
+export default store;
```
Comment (kimnovak) [inline src/redux/store/store.ts]: src/redux got renamed to src/reduxStore, you should delete the old directory

**Review by kimnovak — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (Zola987) [reply src/App.tsx]: I put test page, for Admin check

**Pull request merged**